### PR TITLE
fix:[SRM-14269]: Fix null control_hosts issue.

### PR DESCRIPTION
**Pull request by naveen-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46468)

### CODE REVIEW

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/HostRecordServiceImpl.java [hunk 1]
```diff
@@ -72,9 +73,13 @@ private HostRecord toHostRecord(HostRecordDTO hostRecordDTOs) {
     return HostRecord.builder()
         .accountId(hostRecordDTOs.getAccountId())
         .verificationTaskId(hostRecordDTOs.getVerificationTaskId())
-        .hosts(hostRecordDTOs.getHosts())
+        .hosts(filterHosts(hostRecordDTOs.getHosts()))
```
Comment (dhruv-harness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/HostRecordServiceImpl.java:76]: nit: We can rename the function to filterNullHosts or something similar

**Review by dhruv-harness — APPROVED** [review]

**Review by AbhijithHarness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46468)* status: 1. unresolved discussions. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/HostRecordServiceImpl.java` line 76 at r1](https://reviewable.io/reviews/harness/harness-core/46468#gh-1169494926:-NTIVY0xDwdyI-XLcrlc:b-65rcp0) ([raw file](https://github.com/harness/harness-core/blob/56d969d0d1b2083338ccd7c0bd732b0f92bacf86/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/HostRecordServiceImpl.java#L76)):*
<details><summary><i>Previously, dhruv-harness (Dhruv) wrote…</i></summary><blockquote>

nit: We can rename the function to filterNullHosts or something similar
</blockquote></details>

nit: Ideally we should filter at the time of saving the host Records only.

**Review by AbhijithHarness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46468)* status: :shipit: complete! ✓

**Review by AbhijithHarness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46468)* status: :shipit: complete! ✓

**Pull request merged**